src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs

```diff
@@ -73,34 +70,33 @@ internal sealed partial class SerializableOptionSet : OptionSet
             Debug.Assert(values.Keys.All(ShouldSerialize));
             Debug.Assert(changedOptionKeysSerializable.All(optionKey => ShouldSerialize(optionKey)));
             Debug.Assert(changedOptionKeysNonSerializable.All(optionKey => !ShouldSerialize(optionKey)));
+
+            _languages = new Lazy<ImmutableHashSet<string>>(() => this.GetLanguagesAndValuesToSerialize().languages);
```

Comment:
>languages [](http://example.com/codeflow?start=106&length=9)

Isn't this unnecessary creating the sorted map, since here we only use the language set?

Reply:
yes.  let me revise to not do that.